rpm/ssu.spec

```diff
@@ -20,7 +20,6 @@ Requires(pre): sailfish-setup
 Requires(pre): shadow-utils
 Requires(pre): /usr/bin/groupadd-user
 Requires(postun): shadow-utils
-Requires: ssu-vendor-data
 Requires: ssu-network-proxy
```

Comment:
How can ssu work without vendor data? I think ssu should depend on the vendor data or some kind of similar package.

Reply:
Did you notice the commit message?

Reply:
I noticed the message however ssu won't work without repository information, it doesn't make sense to use ssu without unless some other backends are used in zypper.
Also that it is required in the patterns is nice however the context is missing for anyone outside of whom who can read the patterns.

In the end it's all about semantics I think.

Reply:
Does it fail loudly enough to abort image creation if there's no config supplied? It's silly that the example config was brought in instead, but having no ssu config to install could be considered 'broken' enough for imaging to fail.

Reply:
I am not sure whether it is desired to abort image creation. The task was created in reaction to situation where an image was build in early phase of development, the image could be flashed, the device can be basically used, just the infra was not yet in a shape for ssu use and so the example configs were used, which confused people, ... long story discussed elsewhere. So it can be actually considered an advantage, that it does not hold you back during the early development.

Reply:
BTW the ssu configuration "inside" is not used during image creation. The repos definition comes from the ks.

Reply:
> infra was not yet in a shape for ssu use and so the example configs were used, which confused people, ... long story discussed elsewhere. So it can be actually considered an advantage, that it does not hold you back during the early development.

Ok I get that but exactly that is the issue when changes such as these are created and later down the line there's no context for these changes.

> BTW the ssu configuration "inside" is not used during image creation. The repos definition comes from the ks.

Which itself is generated from ssu config, only additional repos come from the template itself after the ks has been generated.

Reply:
> Ok I get that but exactly that is the issue when changes such as these are created and later down the line there's no context for these changes.

That's not exactly this case, because review was explicitly required from people that have been involved and only those people have been involved that have the right assignment as it goes nowadays you know... The context is distributed among bz, matrix and PR as usually and the discussion is ongoing. Once a conclusion is made, it will definitely be summarized under bz.

> > BTW the ssu configuration "inside" is not used during image creation. The repos definition comes from the ks.
> 
> Which itself is generated from ssu config, only additional repos come from the template itself after the ks has been generated.

They have been able to successfully make the image, they just can't install more software after it is flashed. I haven't checked the details but I doubt it was thanks to the example config.